fix: handle redis connection errors

SassNinja opened the pull request:
### Summary

(If you have not already please refer to the contributing guideline as [described
here](https://github.com/feathersjs/feathers/blob/master/.github/contributing.md#pull-requests))

- [x] Tell us about the problem your pull request is solving.
- [x] Are there any open issues that are related to this?
- [ ] Is this PR dependent on PRs in other repos?

If so, please mention them to keep the conversations linked together.

### Other Information

This PR fixes the bad or rather missing error handling of this hook when connection to redis is not possible.
First of all an error event listener was missing what threw an error if redis is not available. But also the hooks itself didn't catch the case.

What I need is:
- don't throw error if redis connection not possible (either on startup or at runtime)
- reconnect if redis is available
- don't block requests if redis not available and return uncached response instead

Therefore I've implemented a cache strategy, catch it in the hooks (`if (!client) {}`) and added some console.log
Tests are included as well.

Closes https://github.com/idealley/feathers-hooks-rediscache/issues/58

@idealley it would be great if you can review this on the weekend, too. 

If both of my PRs are merged I'm able to use your hooks in production in a much much better way :)

SassNinja committed:
fix: handle redis connection errors

Instead of throwing an error I've implemented retry_strategy and an interval to check if redis is (again) available. If not the service will return uncached response.

SassNinja committed:
remove trailing space

idealley commented:
Could you check please, by pulling the new master? Tests are not passing on my machine with latest master, Travis as well. 

Thank you for all the work.

Sam

SassNinja commented:
You're right, I think due to my adjustments of the redisClient (not immediately available anymore) some tests fail.
I'll fix it (will wait until client is ready).

idealley commented:
Do you have a status on this?

SassNinja commented:
Sorry, haven't had time yet to fix it (lot of work currently)

Will take care of it by the end of the week!

SassNinja committed:
fix: immediately set redisClient

This makes sure you're immiediately able to access the redisClient object and don't need to wait for the ready event.

SassNinja commented:
@idealley it took me some time to understand your tests, but I've been able to fix the issue.
The tests are running again.

Would be great if you'll review my both PRs now and let me know what you think.

SassNinja committed:
improve test coverage

This also tests an offline redisClient in before & after hook.

idealley commented:
Thank you, looks good to me!
